modules/imaging_browser/php/feedback_mri_popup.class.inc

```diff
@@ -0,0 +1,309 @@
+     *
+     * @throws \NotFound If the session id is non-numerical
+     * @throws \LorisException  If the session is not found
+     *
+     * @return void
+     */
+    public function loadResources(
+        \User $user, ServerRequestInterface $request
+    ) : void {
+
+        $path    = $request->getUri()->getPath();
+        $matches = [];
+
+        if (preg_match('#fileID=(\d+)#', $path, $matches) === 1) {
+            $file_id = intval($matches[1]);
+            try {
+                new \LORIS\Image($file_id);
+                $this->file_id = $file_id;
+            } catch (\LorisException $e) {
+                throw new \NotFound("Invalid FileID");
+            }
+        } elseif (preg_match('#sessionID=(\d+)#', $path, $matches) === 1) {
+            $session_id = intval($matches[1]);
+            try {
+                \NDB_Factory::singleton()->timepoint(
+                    new \SessionID(strval($session_id))
+                );
+                $this->session_id = $session_id;
+            } catch (\LorisException $e) {
+                throw new \NotFound("Invalid SessionID");
+            }
+        } else {
+            throw new \NotFound("Invalid Parameters");
+        }
+
+    }
+
+    /**
+     * An NDB_Page can act as both a middleware, or a handler, depending on the
+     * context. When treated as a handler, an NDB_Page calls display() and
+     * converts the output to a PSR7 stream.
+     *
+     * @param ServerRequestInterface $request The PSR15 Request being handled
+     *
+     * @return ResponseInterface The PSR15 response for the page.
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $this->logger->debug("NDB_Page handle called");
+
+        $this->setup();
+
+        if ($request->getMethod() == 'POST') {
+            $this->saveData($request->getParsedBody());
+        }
+
+        return (new \LORIS\Http\Response())
+            ->withBody(new \LORIS\Http\StringStream($this->display() ?? ""));
+    }
+
+    /**
+     * Determine whether the user has permission to view the imaging_browser page
+     *
+     * @param \User $user User object
+     *
+     * @return bool whether the user hass access
+     */
+    function _hasAccess(\User $user): bool
+    {
+        return $user->hasAnyPermission(
+            [
+                'imaging_browser_view_allsites',
+                'imaging_browser_phantom_allsites',
+                'imaging_browser_phantom_ownsite',
+            ]
+        ) || (
+            $user->hasStudySite()
+            && $user->hasPermission('imaging_browser_view_site')
+        );
+    }
+
+    /**
+     *  Gets data for the feedback popup template
+     *
+     * @return array ($subjectData) with data for the template
+     */
+    function getData() : array
+    {
+        $DB       = $this->loris->getDatabaseConnection();
+        $user     = \User::singleton($_SESSION['State']->getUsername());
```

Comment:
I don't understand the `$_SESSION['State']` part of this.. why does that need to be there?

Reply:
Also I think you can get the user out of the $request object in your handle function and use it from there.

Reply:
Actually it would be even better to store it in a class variable in `loadResources`, since it's a resource you need for handling the request. It's even passed as an argument there so you don't need to get it from the request object..

Reply:
ok. Done in the next commit.